p2p/net/swarm/swarm.go

```diff
@@ -602,6 +595,38 @@ func isBetterConn(a, b *Conn) bool {
 	return true
 }
 
+func connPriority(c *Conn) int {
+	if c == nil {
+		return 0
+	}
+
+	var priority int
+	// LAN > WAN > PROXY
+	switch {
+	case c.conn.Transport().Proxy():
+		priority += 10
+	case manet.IsPrivateAddr(c.RemoteMultiaddr()):
+		priority += 20
+	case manet.IsPublicAddr(c.RemoteMultiaddr()):
+		priority += 15
+	}
+
+	// We prefer udp protocols
+	switch c.ConnState().Transport {
+	case "quic", "quic-v1", "webtransport", "webrtc":
+		priority += 5
+	case "tcp", "websocket":
+		priority += 3
+	}
```

Comment:
- `quic` is now unsupported since `quic-go` `v0.37.0`.

---

That looks somewhat fragile to me, what if I run a transport that not on there, altho this can be fixed later.
It should be more granular:
- `quic-v1` > `webtransport` (is a direct downgrade as webtransport is `quic-v1` plus other stuff) > `webrtc` (relies on hacks to function properly due to lack of proper stream close)

Reply:
Thank you for your correction. It is indeed necessary to make a more fine-grained distinction between protocols.